scipy/interpolate/tests/test_interpolate.py

```diff
@@ -753,13 +753,21 @@ def test_read_only(self):
 
 
 class TestLagrange:
-
     def test_lagrange(self):
-        p = poly1d([5,2,1,4,3])
+        p = poly1d([5, 2, 1, 4, 3])
         xs = np.arange(len(p.coeffs))
         ys = p(xs)
-        pl = lagrange(xs,ys)
-        assert_array_almost_equal(p.coeffs,pl.coeffs)
+        pl = lagrange(xs, ys)
+        assert_array_almost_equal(p.coeffs, pl.coeffs)
+
+    def test_reduced_coeffs(self):
+        a = np.array([0, 2, 1])
+        p = poly1d(a)
+        x = np.arange(len(a))
+        y = p(x)
+        pl = lagrange(x, y, reduce=False)
+        assert_array_almost_equal(a, pl[1])
+        assert_array_almost_equal(p.coeffs, pl[0].coeffs)
```

Comment:
Nit: in new code, prefer `assert_allclose` with explicit tolerance (`atol=1e-14` or some such for testing fp equality)